sound/soc/intel/atom/sst/sst_stream.c

```diff
@@ -217,7 +217,7 @@ int sst_send_byte_stream_mrfld(struct intel_sst_drv *sst_drv_ctx,
 		 * copy the reply and send back
 		 * we need to update only sz and payload
 		 */
-		if (bytes->block) {
+		if (bytes->block && block) {
```

Comment:
As a comment on how static analyzers work (GCC or others), they _cannot_ reliably track null checks on anything that isn't a local variable, because of the chance of concurrent modification (whether from a signal, interrupt in kernel, or from a different core). From this point of view it's not a false positive. It's not something that a new version of the analyzer will fix.

The correct fix for such issues is to grab a copy of the pointer locally, and do the NULL check and usage on said local copy. If you know for a different reason that the pointer in the struct cannot be concurrently modified you can probably ignore the report as well.

This _specific_ fix isn't exactly to my liking, because it's bypassing the checker without really fixing what it points out.

Reply:
fixed with local variable as suggested